src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -324,6 +324,21 @@ public void createControlledResource(ControlledResource controlledResource)
           "No cloud context found in which to create a controlled resource");
     }
 
+    // scan entire attributes column for existing controlled resource
+    // TODO: use a dedicated URI column to make this more efficient PF-690
+    final String attributesSql =
+        String.format(
+            "SELECT attributes FROM resource "
+                + "WHERE resource_type = '%s' "
+                + "AND stewardship_type = 'CONTROLLED'",
+            controlledResource.getResourceType().toSql());
+    final List<String> attributesList =
+        jdbcTemplate.queryForList(attributesSql, Collections.emptyMap(), String.class);
+    final boolean exists =
```

Comment:
nit: seems like excessive use of `final` on local variables. (ditto above). I'm not sure we need extra characters to signal we're not changing these variables on a short function where we have no reason to think otherwise. go/java-practices/final#params_locals